controllers/nginx/rootfs/etc/nginx/template/nginx.tmpl

```diff
@@ -236,7 +236,19 @@ http {
         ssl_verify_client on;
         ssl_verify_depth {{ $location.CertificateAuth.ValidationDepth }};
         {{ end }}
+        
+        {{ if (or $location.Redirect.ForceSSLRedirect (and (not (empty $server.SSLCertificate)) $location.Redirect.SSLRedirect)) }}
```

Comment:
By the way, acording to NGINX docs, the IF and redirects can happen inside a location, a if and a server directive. The usage of this directive inside the 'location' limits the redirection only for the current location (maybe only the '/' path), not containing the other paths that might exist.

I'm on vacation (and traveling) right now, but will take a look again in this behaviour ASAP.

@jasonbrownbridge How have you detected this? Can you please reproduce the taken steps, so I can try here and take a look.